File: mcr_robinson_hello_world/common/src/HelloWorldImpl.cpp
#include "mcr_robinson_hello_world/HelloWorldImpl.h"

namespace mcr_robinson {
namespace helloworld {

int counter = 0;

HelloWorldImpl::HelloWorldImpl() {
	std::cout << "HelloWorldImpl()" << std::endl;
}

HelloWorldImpl::~HelloWorldImpl() {
	std::cout << "~HelloWorldImpl()" << std::endl;
}

void HelloWorldImpl::update() {
	std::cout << "HelloWorldImpl::update()" << std::endl;

	std::stringstream ss;
	ss << "Hello " << this->name << " " << counter++ << std::endl;
	std::cout << " HelloWorldImpl sending: " << ss.str() << std::endl;

	dataport_output_answer(ss.str());
	eventport_output_done();

}

bool HelloWorldImpl::init() {
	std::cout << "HelloWorldImpl::init()" << std::endl;
	return true;
}

void HelloWorldImpl::shutdown() {
	std::cout << "HelloWorldImpl::shutdown()" << std::endl;

}

void HelloWorldImpl::dataport_input_name(std::string name) {
	std::cout << "HelloWorldImpl::dataport_input_name()" << std::endl;

	this->name = name;
	std::cout << "HelloWorld got name " << name << std::endl;
}

} /* namespace manipulation */
} /* namespace mir_industrial */

File: mcr_robinson_java/robinson.helloworld/src/main/java/helloworld/HelloWorldImpl.java
package helloworld;


public class HelloWorldImpl extends IHelloWorld {

	
	String name;
	
	int counter = 0;
	
	public void pin_name(String name) {
		System.out.println("HelloWorld:pin_name: " + name);
		this.name = name;
	}
	
	public void event_in_hello() {
		System.out.println("HelloWorld:event_in_hello");
	}
	
	public void update() {
		System.out.println("HelloWorld:update");
		
		this.pout_answer.emit("Hello " + this.name + " - " + counter++);
		
		this.event_out_done.emit();
	}
	
	public void init() {
		System.out.println("HelloWorld:init");
	}
	
	public void shutdown() {
		System.out.println("HelloWorld:shutdown");
	}
	
}
